plenum/server/consensus/monitoring/primary_connection_monitor_service.py

```diff
@@ -123,6 +122,16 @@ def _check_primary_connection(self):
         self._propose_view_change()
         self._propose_view_change_timer.start()
 
+    def _is_primary_connected(self):
+        if self._data.is_primary:
+            return True
+
+        primary_name = replica_name_to_node_name(self._data.primary_name)
+        if primary_name in self._network.connecteds:
+            return True
+
+        return primary_name not in self._data.validators
```

Comment:
Is it really correct? What if primary is known (that is `self._data.validators` contains `primary_name `), but it's just not connected? Previous version of code would return False (not connected), but the current code will return True.

Reply:
If primary is known (primary not in validators = False), but not connected this code will return False as well. Edge case that wasn't captured before this code is situation when due to some out of sync condition we have a primary, which is not yet in known validator set. I have a hard time finding out how can this be possible, but there was old unit test for specifically this case, so I ported it to new tests for PrimaryConnectionMonitorService.